File: Rubik.HTML/Server/Schema/CellSet.cs
using System;
using System.Web;
using System.ComponentModel;
using System.Collections.Generic;

namespace Rubik.Server.Schema
{
    public class CellSet
    {
        public List<Cell> Cells
        {
            get; set;
        }

        public ResultAxis Columns
        {
            get; set;
        }

        public ResultAxis Rows
        {
            get; set;
        }

        public ResultAxis Filters
        {
            get; set;
        }
    }
}

File: Rubik.HTML/Lib/DataHub/Schema/CellSet.ts
module Rubik.DataHub {
    export interface ICellSet {
        cells: Rubik.Collections.IObjectDictionary<ICell>;
        columns: IResultAxis;
        rows: IResultAxis;
        filters: IResultAxis;
    }

    export interface IResultAxis {
        hierarchies: IResultHierarchy[];
        positions: IPosition[];
    }

    export interface ICell {
        value: any;
        fmtValue: string;
    }

    export interface IPosition {
        ordinal: number;
        members: IResultMember[];
    }

    export interface IResultMember {
        caption: string;
        uniqueName: string;
        childCount: number;
        levelName: string;
        levelDepth: number;
        drilledDown: boolean;
        parentSameAsPrevious: boolean;
        properties: IMemberProperty[];
    }

    export interface IResultHierarchy {        
        uniqueName: string;        
        index: number;      
        properties: IHierarchyProperty[];
    }

    export interface IHierarchyProperty {
        name: string;
        uniqueName: string;        
    }
    export interface IMemberProperty {
        name: string;
        uniqueName: string;
        value: any;
    }
}